### ISSUE TITLE
New ML hobbyist

### ISSUE THREAD

**Issue by wmeitzen** [maximecb/gym-miniworld#19]
Hi,

ML noob here.  I've been looking at different projects on the web, mostly for fun and to learn something new.

I love your gym-miniworld project b/c I can see its progress (cones, wall, sidewalk, a building in the distance, etc.)

Please see the screenshot.  Does "success rate" > 0 mean it's touched the block?  I've been running gym-miniworld all night, and I've seen a success rate of 0.01 or 0.02 only a few times.

Because of its low success rate, I don't know if I did something wrong, or if I should be more patient.  What should I expect?

Thank you!

Edit: I entered this command to start:
```bash
administrator@ubuntu:~/Desktop/gym-miniworld/pytorch-a2c-ppo-acktr$` python3 main.py --algo ppo --num-frames 5000000 --num-processes 16 --num-steps 80 --lr 0.00005 --env-name MiniWorld-Sidewalk-v0
```

![miniworld-ss](https://user-images.githubusercontent.com/13643084/73073422-ea89fd00-3e7c-11ea-9422-aaa625e1c421.jpg)

**Comment by maximecb** [comment]
Hi William,

The success rate is the proportion of episodes with a reward greater than zero, so the proportion in which the agent touched the cube.

What parameters are you running the training code with? Usually you want 16 processes. Different environment might work better with different learning rates as well.

**Comment by wmeitzen** [comment]
Hi maximecb,

Here are the parameters I'm using:

```bash
python3 main.py --algo ppo --num-frames 5000000 --num-processes 16 --num-steps 80 --lr 0.00005 --env-name MiniWorld-Sidewalk-v0
```

**Comment by maximecb** [comment]
My next questions would be: are you running this on a machine with a GPU? How far along in the training did you get (how many millions of frames)?

You can check if a GPU is available with
`
python3
import torch
print(torch.cuda.is_available())
`

If you are running this on a machine with a GPU and training did get far without converging, are you able to solve easier environments? The easiest would be `MiniWorld-Hallway-v0`. I was able to train that to convergence within 20 minutes or so on my desktop.

**Comment by wmeitzen** [comment]
The GPU check returns "False".

I got about 3.1M timesteps, but I don't know if that answers your training question.  Maybe the screenshot from my initial message will.

Here was my Sidewalk environment command:
```bash
administrator@ubuntu:~/Desktop/gym-miniworld/pytorch-a2c-ppo-acktr$` python3 main.py --algo ppo --num-frames 5000000 --num-processes 16 --num-steps 80 --lr 0.00005 --env-name MiniWorld-Sidewalk-v0
```

I'll try the Hallway one next.

P.S. Thanks for your help!

**Comment by maximecb** [comment]
If you're training without CUDA it's going to be much slower. Do you have a GPU on your machine?

You'd want to install the version of PyTorch explicitly built with cuda support, and also the nvidia cuda toolkit.

**Comment by wmeitzen** [comment]
I don't have a GPU installed.  FWIW, it's a pretty decent machine: Intel i7 @ 4GHz, 32G RAM, 250G SSD, 2TB HD.

Oddly, after about an hour of running the Sidewalk environment, I saw what looked like success from the command line:
```bash
Updates 180, num timesteps 231680, FPS 68 
 Last 100 training episodes: mean/median reward 0.97/0.98, min/max reward 0.94/1.00, success rate 1.00
```

... but `enjoy.py` didn't look like it had succeeded.  On a whim, I stopped and restarted `enjoy.py`, and the agent navigated to the block pretty consistently.

I appreciate all your help!  All my best.
